packages/client-sdk-nodejs/test/integration/auth.test.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  GenerateApiToken,
+  MomentoErrorCode,
+  RefreshApiToken,
+} from '@gomomento/sdk-core';
+import {SetupAuthIntegrationTest} from './integration-setup';
+
+const {Momento, SessionToken} = SetupAuthIntegrationTest();
+
+describe('Integration tests for generting api tokens', () => {
+  it('should succeed for generating an api token that expires', async () => {
+    const secondsSinceEpoch = Math.round(Date.now() / 1000);
+    const expireResponse = await Momento.generateApiToken(SessionToken, 10);
+    const expiresIn = secondsSinceEpoch + 10;
+
+    expect(expireResponse).toBeInstanceOf(GenerateApiToken.Success);
+
+    const expireResponseSuccess = expireResponse as GenerateApiToken.Success;
+    expect(expireResponseSuccess.is_success);
+
+    expect(expireResponseSuccess.validUntil).toBeWithin(
+      expiresIn - 1,
+      expiresIn + 1
+    );
+  });
+
+  it('should succeed for generating an api token that never expires', async () => {
+    const neverExpiresResponse = await Momento.generateApiToken(
+      SessionToken,
+      undefined
+    );
+    expect(neverExpiresResponse).toBeInstanceOf(GenerateApiToken.Success);
+    const neverExpireResponseSuccess =
+      neverExpiresResponse as GenerateApiToken.Success;
+    expect(neverExpireResponseSuccess.is_success);
+    expect(neverExpireResponseSuccess.validUntil).toEqual(undefined);
```

Comment:
The structure of this `success` response is another thing we need to think about the API for.  I suspect that we will want to use the same `ExpiresAt` or `ExpiresIn` type of object as I linked to in one of my comments above, as opposed to having `undefined` imply "never expires".  We can discuss this if people have thoughts but we have to get it right before we ship this.